eng/pipelines/templates/stages/archetype-sdk-client.yml

```diff
@@ -59,7 +59,10 @@ stages:
       parameters:
         ServiceDirectory: ${{ parameters.ServiceDirectory }}
         DependsOn:
-        - Build
+          - Build
+          # Only depend on `LiveTest` if there are live tests to execute
+          - ${{ if and(eq(variables['System.TeamProject'], 'internal'), ne(parameters.LiveTestCtestRegex, '')) }}:
```

Comment:
Is this needed? If the LiveTests step has the same runtime condition won't it essentially be skipped but be marked as completed so this will continue?

Reply:
Live tests are included only if requirements for live tests (ctest regex) are present. If not then the stage is not created at the level of the template. 

If you include a stage which does not exist you'll get an error when queuing a build against a pipeline that does not have live tests (e.g. `cpp - template`) 

![image](https://user-images.githubusercontent.com/2158838/123716378-4b1a2800-d82f-11eb-89ac-d4d1aa50a84f.png)

Reply:
Ok you would need to turn that into a runtime condition then to take that approach. Your approach works as well but it does require we duplicate this template condition here and in the test pipeline.

Reply:
Opened https://github.com/Azure/azure-sdk-for-cpp/issues/2520 to track.